mountpoint-s3/src/upload.rs

```diff
@@ -136,9 +149,25 @@ impl<Client: ObjectClient> UploadRequest<Client> {
     pub async fn complete(self) -> Result<PutObjectResult, PutRequestError<Client>> {
         let size = self.size();
         let checksum = self.hasher.finalize();
-        self.request
+        let result = self
+            .request
             .review_and_complete(move |review| verify_checksums(review, size, checksum))
-            .await
+            .await?;
+        self.sse
+            .verify_response(result.sse_type.as_deref(), result.sse_kms_key_id.as_deref())
+            .unwrap_or_else(|err| {
+                error!(
+                    "SSE settings were corrupted, object {} WAS uploaded to S3, terminating, error: {}",
+                    self.key, err
+                );
+                // Reaching this point is very unlikely and means that SSE settings were corrupted in transit or on S3 side, this may be a sign of a bug
+                // in CRT code or S3. Thus, we terminate Mountpoint to send the most noticeable signal to customer about the issue. We conciously choose
+                // exiting instead of returning an error because:
```

Comment:
typo: consciously

But maybe reword like: 
```suggestion
                // in CRT code or S3. Thus, we terminate Mountpoint to send the most noticeable signal to customer about the issue.
                // We prefer exiting instead of returning an error because:
```